staging/src/k8s.io/component-base/metrics/prometheus/controllers/metrics.go

```diff
@@ -42,14 +42,15 @@ type ControllerManagerMetrics struct {
 
 // NewControllerManagerMetrics create a new ControllerManagerMetrics, with specific manager name.
 func NewControllerManagerMetrics(manager string) *ControllerManagerMetrics {
+	register()
```

Comment:
I am not a fan of this approach, `NewControllerManagerMetrics` should only create a ControllerManagerMetrics object if we follow the usual constructor pattern.

I would prefer for the exported Register() to be called after the instantiation of a ControllerManagerMetrics object.

Reply:
I understand the sentiment from the reader perspective to have a clean meaning behind a fuction.

Added calls to `Register` after `NewControllerManagerMetrics` call.